src/beast/examples/http_sync_server.h

```diff
@@ -94,23 +94,36 @@ class http_sync_server
         }
     }
 
+    struct lambda
+    {
+        int id;
+        http_sync_server& self;
+        socket_type sock;
+        boost::asio::io_service::work work;
+
+        lambda(int id_, http_sync_server& self_,
+                socket_type&& sock_)
+            : id(id_)
```

Comment:
I don't think so, the caller owns the global variable so they need to increment it before passing it in